### ISSUE TITLE
zen_mail: Storefront mailing attempted when 'Send Emails' is disabled

### ISSUE THREAD

**Issue by lat9** [zencart/zencart#1850]
When a store (during testing) has been configured to not send emails, e.g. `Configuration->Email Options->Send Email` is set to **false**, emails are still being attempted.  That results in a slew of unwanted debug-logs (both storefront and admin):

`PHP Notice:  Email Error: Could not instantiate mail function.`

**Comment by drbyte** [comment]
That must be from non-core code, ie: direct calls to PHP's `mail()` function.

The 2nd line of `zen_email()` immediately aborts, so it never calls `mail`:
```php
      if (SEND_EMAILS != 'true') return false;  // if sending email is disabled in Admin, just exit
```

**Comment by drbyte** [comment]
Argh. I don't understand why that first line is even there. I probably put it there, but I know I never use it.

Strongly leaning towards removing that first condition.

**Comment by lat9** [comment]
That was my initial thought as well, but I didn't know why that initial clause is present.

**Comment by drbyte** [comment]
Looks like it was added in v1.3.7.
And an interim step which got refactored was:
```php
    if (SEND_EMAILS != 'true' && defined('DEVELOPER_OVERRIDE_EMAIL_STATUS') && DEVELOPER_OVERRIDE_EMAIL_STATUS != 'send') return false;  // if sending email is disabled in Admin, just exit
```

I'm going to just remove it.

Although, you shouldn't be triggering it unless `DEVELOPER_OVERRIDE_EMAIL_STATUS` is actually defined for you.

**Comment by drbyte** [comment]
Maybe I'll leave it, but make it allow configuring the abort, like this:

```php
if (defined('DEVELOPER_OVERRIDE_EMAIL_STATUS') && DEVELOPER_OVERRIDE_EMAIL_STATUS == 'false') return false;  // disable email sending when in developer mode
```

**Comment by lat9** [comment]
FWIW, I don't have `DEVELOPER_OVERRIDE_EMAIL_STATUS` defined for the site that's throwing those debug-logs.

**Comment by drbyte** [comment]
Closed by #1850
